src/socialMediaActivity.js

```diff
@@ -822,46 +1011,69 @@ function findFieldFacebook(object, fieldName, enterArray = true, recurseLevel =
+        details.eventTime = eventTime;
+        details.postAudience = checkFacebookPostAudience(requestDetails);
+        details.reshareSource = await getReshareInfo();
         const variables = findFieldFacebook(requestDetails.requestBody.formData, "variables");
         const message = findFieldFacebook(variables, "message");
-        details.newPostMessage = message ? findFieldFacebook(message, "text") : "";
-        details.attachedUrls = [];
+        details.postText = message ? findFieldFacebook(message, "text") : "";
+        details.postAttachments = [];
         try {
             const attachments = findFieldFacebook(variables, "attachments");
             const link = findFieldFacebook(attachments, "link");
             const canonical = findFieldFacebook(link, "canonical");
-            details.attachedUrls.push(canonical);
+            if (canonical !== null) {
+                details.postAttachments.push(canonical);
+            }
+            // Facebook's post IDs are extremely large integers, stored as such in JSON.
+            // JSON doesn't support parsing such large values, and will instead round them
+            // to close approximations (which of course fails to give us accurate values for
+            // the post ID), so we use regexes to extract the relevant values without
+            // JSON parsing the string.
+            const shareData = link.share_scrape_data;
+            const shareTypeRegex = /"share_type":(\d+)/;
+            const shareParamsRegex = /"share_params":\[(\d+)\]/;
+            const shareType = shareTypeRegex.exec(shareData);
+            if (shareType != null && shareType.length == 2 && shareType[1] == "99") {
+                const sharedPostId = shareParamsRegex.exec(shareData);
+                if (sharedPostId != null && sharedPostId.length == 2) {
+                    details.actedUponPostId = sharedPostId[1];
+                }
+            }
         } catch {
             debugLog("failed extracting links from facebook content");
         }
-        const audience = checkFacebookPostAudience(requestDetails);
-        const source = await getReshareInfo();
-        details.audience = audience;
-        details.source = source;
-        details.eventType = "reshare";
-        details.eventTime = eventTime;
         return details;
     }
 }
 
 /**
+ * Ask our Facebook content script whether the last reshare was from a page or a person.
+ * This information is not included in the web request, nor is it attached to the generated post,
+ * so the only way to get it is by watching the user click the reshare button, which
+ * the content script does.
+ * @returns {string} - The source of the last reshare ("person" or "page").
  * @private
  */
-async function getReshareInfo() {
+function getReshareInfo() {
     return browser.tabs.sendMessage(facebookTabId, {"recentReshare": true}).then((response) => {
         return response;
     }, (e) => { console.log("ERROR", e); } );
 }
```

Comment:
Would use the `messaging` module and a schema. Would also give a more descriptive error.